Add docs for the image pull secrets CO option (#1581)
strimzi/strimzi-kafka-operator: documentation/book/con-customizing-pods.adoc

|Defines a list of references to {ProductPlatformName} Secrets that can be used for pulling container images from private repositories.
For more information about how to create a Secret with the credentials, see {K8sPullingImagesFromPrivateRegistries}.

> NOTE: When both the `STRIMZI_IMAGE_PULL_SECRETS` environment variable in Cluster Operator and `imagePullSecrets` option are specified, only the `imagePullSecrets` will be used and the `STRIMZI_IMAGE_PULL_SECRETS` will be ignored.

Reviewer: ```suggestion
NOTE: When the `STRIMZI_IMAGE_PULL_SECRETS` environment variable in Cluster Operator and the `imagePullSecrets` option are specified, only the `imagePullSecrets` variable is used. The `STRIMZI_IMAGE_PULL_SECRETS` variable is ignored.
```
